packages/rust/server/tests/user_test.rs

```diff
@@ -0,0 +1,301 @@
+use crate::helper::start_server;
+use axum::http::StatusCode;
+use bubble::models::confirmation::Confirmation;
+use bubble::models::user::User;
+use bubble::routes::user::{Confirm, CreateUser};
+
+use sqlx::{Executor, Row};
+
+mod helper;
+
+#[tokio::test]
+async fn create_user() {
+    let (db, client) = start_server().await;
+
+    let _clean = cleanup!(|db| {
+        db.execute("DELETE FROM \"confirmation\"").await.unwrap();
+        db.execute("DELETE FROM \"session_token\"").await.unwrap();
+        db.execute("DELETE FROM \"user\"").await.unwrap();
+    });
+
+    let res = client
+        .post("/user/signup")
+        .header("Content-Type", "application/json")
+        .body(
+            serde_json::to_string(&CreateUser {
+                email: "test@gmail.com".to_string(),
+                username: "test".to_string(),
+                password: "password".to_string(),
+                phone: Some("18001239876".to_string()),
+                name: "John Doe".to_string(),
+            })
+            .unwrap(),
+        )
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::CREATED);
+
+    //TEMP
+    let user_uuid = User::uuid_from_username(&db, "test").await.unwrap();
+    let user = User::from_uuid(&db, user_uuid)
+        .await
+        .expect("user doesn't exist");
+
+    assert_eq!(user.username, "test");
+    assert_eq!(user.password, "password");
+    assert_eq!(user.profile_picture, None);
+    assert_eq!(user.email, None);
+    assert_eq!(user.phone, Some("18001239876".to_string()));
+    assert_eq!(user.name, "John Doe");
+
+    let row = sqlx::query("SELECT * FROM confirmation WHERE id = 1;")
+        .fetch_one(&db)
+        .await
+        .unwrap();
+
+    let conf = Confirmation {
+        id: row.get("id"),
+        user_id: row.get("user_id"),
+        link_id: row.get("link_id"),
+        email: row.get("email"),
+        created: row.get("created"),
+    };
+
+    assert_eq!(conf.user_id, user.id);
+    assert_eq!(conf.email, "test@gmail.com");
+
+    let confirm_res = client
+        .post("/user/signup-confirm")
+        .header("Content-Type", "application/json")
+        .body(
+            serde_json::to_string(&Confirm {
+                link_id: conf.link_id.to_string(),
+            })
+            .unwrap(),
+        )
+        .send()
+        .await;
+
+    assert_eq!(confirm_res.status(), StatusCode::CREATED);
+
+    let user = User::from_uuid(&db, user_uuid).await.unwrap();
+
+    assert_eq!(user.username, "test");
+    assert_eq!(user.password, "password");
+    assert_eq!(user.profile_picture, None);
+    assert_eq!(user.email, Some("test@gmail.com".to_string()));
+    assert_eq!(user.phone, Some("18001239876".to_string()));
+    assert_eq!(user.name, "John Doe");
+}
+
+#[tokio::test]
+async fn create_multiple_user() {
+    let (db, client) = start_server().await;
```

Comment:
If you want to do this, do it in a loop.

Reply:
Is the current state of the test acceptable? I intended to test signing up and confirming users in varying orders to ensure nothing conflicts.